au4/src/projectscene/view/clipsview/timeformat.h

```diff
@@ -0,0 +1,25 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+**********************************************************************/
+
+#pragma once
+
+#include "rulerformat.h"
+
+class TimeFormat final : public RulerFormat
```

Comment:
At the moment, the reasons why цу need to have a base class with virtual methods and an inheritor are not clear. I believe this is not necessary.

Reply:
And I think these methods can be made static for convenience

Reply:
It makes sense if you know that there can be different types of rulers in Audacity. I implemented timeline but you can also have "Beats and measures" ruler - which needs to have the same methods implemented. I can remove it right now if you want but we're gonna need it later anyway

Reply:
Yes, I assumed that there might be something like that, but then the architecture must be appropriate. 
that is, then it turns out that there should be something like this:
* Single implementation of timeline for different things
* Must be type timeline 
* There must be a factory that created a formatter for a given type.

(These are all assumptions, I didn't look at how it was done in AU3 :) )

Yes, everything may be like this... but it may not be like that :) Therefore, I suggest not complicating things until it is clear why it is needed. Because in reality it may be a little different.

Reply:
Sure, then I'm going to remove base class for now